src/app/utils/api/matches.ts

```diff
@@ -0,0 +1,37 @@
+import { createPostOptions, fetchJSON } from '.';
+import { MatchClient, PaginatedMatchesClient } from '../../../types';
+import { getCurrentUser } from '../user';
+
+export async function getMatches(query: {
+  page: number;
+  itemsPerPage: number;
+}): Promise<PaginatedMatchesClient> {
+  const { page, itemsPerPage, ...rest } = query;
+  const searchParams = new URLSearchParams({
+    page: page.toString(),
+    itemsPerPage: itemsPerPage.toString(),
+    ...rest,
+  });
+  const queryString = searchParams.toString();
+
+  const matches = await fetchJSON<PaginatedMatchesClient>(
+    `/api/matches?${queryString}`
+  );
+  matches.results = matches.results.map((match) => ({
+    ...match,
+    createdAt: new Date(match.createdAt),
+  }));
+  return matches;
+}
+
+export async function postMatch(gameId: number): Promise<MatchClient> {
+  const { username } = (await getCurrentUser()) || { username: 'unknown' };
```

Comment:
I think the a match requires the `avatarSrc` too. Not important to add it to this pull request, just noticed.